src/include/simeng/Core.hh

```diff
@@ -33,6 +33,10 @@ class Core {
 
   /** Retrieve a map of statistics to report. */
   virtual std::map<std::string, std::string> getStats() const = 0;
+
+  /** Change the value of the ARM64_SYSREG_CNTVCT_EL0 system register to number
+   * of cycles completed.*/
+  virtual void incCNTVCT(uint64_t iterations) = 0;
```

Comment:
Any chance this could not be in the base core header file? Somewhere in the architecture file maybe. This file should be ISA agnostic...